giraph-core/src/main/java/org/apache/giraph/job/CombinedWorkerProgress.java

```diff
@@ -204,4 +215,19 @@ public String toString() {
     }
     return sb.toString();
   }
+
+  /**
+   * Check if this instance made progress from another instance
+   *
+   * @param lastProgress Instance to compare with
+   * @return True iff progress was made
+   */
+  public boolean madeProgressFrom(CombinedWorkerProgress lastProgress) {
```

Comment:
Why not use the underlying raw numbers instead of the string? For instance, small changes in memory may not really mean progress.